app/controllers/tickets_controller.rb

```diff
@@ -189,4 +189,10 @@ def anonymize
       new_wca_id: person&.reload&.wca_id, # Reload to get the new wca_id
     }
   end
+
+  def imported_temporary_results
+    competition = Competition.find(params.require(:competition_id))
+
+    render json: competition.inbox_results.includes(:inbox_person)
```

Comment:
Have you actually tried out whether this works? The scope `inbox_person` depends on a composite key defined with a `-> () { where() }` style block, Rails usually isn't able to preload these on the fly

Reply:
Surprisingly, this works. No idea what kind of lazy-loading Rails internal black magic is behind this, but I'll take it as a win. Kudos for figuring that out!